build-sonarlint/downloadUtil.mjs

```diff
@@ -0,0 +1,29 @@
+/* --------------------------------------------------------------------------------------------
+ * SonarLint for VisualStudio Code
+ * Copyright (C) 2017-2025 SonarSource SA
+ * sonarlint@sonarsource.com
+ * Licensed under the LGPLv3 License. See LICENSE.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+
+import artifactory from './artifactory.mjs';
+import { createWriteStream } from 'fs';
+import { Readable } from 'stream';
+
+const HTTP_BAD_REQUEST = 400;
+
+export async function downloadFile(fileUrl, destPath, useAuthentication=false) {
```

Comment:
Extracted to a dedicated util module since both the JRE and O# download rely on it.